explainer.md

```diff
@@ -215,11 +224,206 @@ defaults.
+      // Output with "safe" and "unsafe" methods are the same for this config.
 };
 const config_block = {
-  blockElements: [ "style" ]  // Allows a lot of things. But not <style>.
+  blockElements: [ "style" ],  // Allows a lot of things. But not <style>.
+  blockAttributes: [ "class", "style" ]  // No style or class attribute either.
       // And not XSS-y stuff, either, if used with a "safe" method.
       // Output with "safe" and "unsafe" methods might be quite different.
 };
 ```
+
+### Configuration List-Types: Allow-lists, Blocks-lists, and Drop-lists
+
+For elements there are three types of lists: Allow-lists, Block-lists
+and Drop-lists. These correspond to three related but somehow different use
+cases:
+
+If you have a specific set of functionality in mind that you wish to
+retain and would like to drop everything else as a matter of caution, this
+can be accomplished with allow lists. This is probably the most common case.
+For example, you might want allow user comments with simple styling, but
+no style definitions:
+
+```js
+const config_only_simple_styling = {
+  allowElements: ["p", "span", "em", "b", "i", "u", "s", "img"],
+  allowAttributes: ["class", "src"],
+};
+```
+
+If you wish to let most content pass through, but have a specific set of
+functionality in mind that you wish to also block in your application, you can
+use block-lists. For example, you might want to chiefly drop XSS-y content, but
+also disallow new styles or `id`s not already declared in your document:
+
+```js
+const config_anything_but_style_definitions = {
+  blockElements: ["style"],
+  blockAttributes: ["style", "id"],
+};
+```
+
+If you know your input will contain elaborate markup and you only wish to
+retain some of it, but you want to retain the full textual content, you
+can use `dropElement`. This will remove only elements but retain their children.
+There is no `dropAttribute` because attribute nodes do not have children.
+
+```js
+const config_that_drops_things = {
+  dropElements: ["span", "em", "u", "s", "i", "b"]
+};
+```
+
+### Configuring attributes per element
+
+A common use case is to allow or block all instances of a given attribute,
+but this isn't always sufficient. One may also act on attributes on specific
+elements.
+
+In the example `config_text_with_only_simple_styling` in the previous chapter
+we have allowed the `src` attribute on any of the eight allowed elements.
+If one wanted to allow `class` everywhere, but `src` only on `<img>`, the
+following would do:
+
+```js
+const config_with_element_specific_attributes = {
+  allowElements: [
+    "p", "b", "i",
+    { name: "img", attributes: [ "src" ] }
+  ],
+  allowAttributes: ["class"],
+};
+```
+
+If you want to drop `src` attributes from `<input>` elements but retain them
+elsewhere, you can use:
+
+```js
+const drop_src_attribute_from_input = {
+  dropElements: [{ name: "input", attributes: ["src"]}],
+}
+```
```

Comment:
Should this drop the element or the attribute?

Reply:
Looking back at https://github.com/WICG/sanitizer-api/issues/181, we had `dropElements` just accepting elements and `dropAttributes` to accept `{name: 'src', elements: ['input'] }`, which is a bit inverse of what the allow-properties do